babette: first draft blog post

richelbilderbeek opened the pull request:
As agreed with @stefaniebutland [here](https://github.com/ropensci/software-review/issues/209#issuecomment-453273641), I submit the first draft of my `babette` blog post, as part of the [rOpenSci onboarding process](https://github.com/ropensci/software-review/issues/209).

stefaniebutland commented:
Thank you @richelbilderbeek.
This link shows the preview of your post: https://deploy-preview-415--ropensci.netlify.com/blog/2019/01/06/babette/

I have a couple of structural suggestions based on my first look. I can review the content on Monday Jan 28. 
- Thank you for acknowledging the reviewers. It's also very nice that you thanked people who supplied use cases and bug reports. Would you please link to them, e.g. GitHub or Twitter?
  - I'd like to suggest an alternative wording for citing Guangchuang Yu's contribution: "Reviews were provided by David Wright and Joëlle Barido-Sottani, with a contribution from Guangchuang Yu."
  - you might consider linking to the actual review thread as `[text to link from](https://github.com/ropensci/software-review/issues/209`), but that is optional
  - Soon we will be specifically encouraging people to submit use cases in our discussion forum in a templated format, so it will be great to have those included. I'll follow up when your post is published
- Please include only the images and blog post content themselves, but not e.g. `themes/ropensci/static/img/blog-images/2019-01-06-babette/create_figures.R`. You could include a link in the post to this content in your own repo if you would like to provide that. This might also apply to the xml file, but I have not yet checked the post details to see how it is integrated. Usually you would submit only `.md` and any image files.
- add `!` in front of each `[](img/blog-images/2019-01-06-babette/phylogeny.png)` to link to image
- For references, in text, use [^1] and at bottom of post, use [^1]: blah blah reference
Example: https://ropensci.org/blog/2019/01/22/waterinfo-tidal-eel/#fn:1
Raw .md: https://raw.githubusercontent.com/ropensci/roweb2/master/content/blog/2019-01-22-waterinfo-tidal-eel.md. Link from DOI to the paper
- tag `software-peer-review` -> Software Peer Review and `r` -> `R`
  - very recent format change
- Please create an author's index file for yourself as described [here](https://github.com/ropensci/roweb2#blog-post-editorial-suggestions). This is a new feature in our site.

Please let me know if you would like me to clarify anything.

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+   2|BEAST2 input file|BEAST2   |posterior files
+   3|posterior files  |DensiTree|plot of the phylogenies
+
+ 1. Run BEAUti to create a BEAST2 input file
```
please link to and/or reference BEAUTi and DensiTree

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+Doing these steps manually would be a drag.
+Instead, I created `babette` to work with BEAST2 from R.
+
+#### With `babette`
```
```suggestion
#### With babette
```

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+
+Again, `babette` is a package to work with BEAST2 from R.
+The Bayesian inference of BEAST2 works on an alignment,
+stored in FASTA format. This is how such a FASTA file looks like:
```
```suggestion
stored in FASTA format. This is what such a FASTA file looks like:
```

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+`babette` actually is a suite of five packages, 
+calling and/or offering an alternative for the BEAST2 tool suite:
+
+What|BEAST2 suite|`babette` suite
```
```suggestion
What|BEAST2 suite|babette suite
```

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+Note that for plotting the posterior trees, I've added many arguments to 
+make it a beautiful blog post picture :-)
+
+#### More about `babette`
```
```suggestion
#### More about babette
```

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+Visualize posterior phylogenies|DensiTree|`babette`
+Install BEAST2 packages|BEAUti|`mauricer`
+
+The BEAST2 tool suite has -at the moment- more functionality than `babette`.
```
```suggestion
The BEAST2 tool suite has - at the moment - more functionality than `babette`.
```
add spaces

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+Install BEAST2 packages|BEAUti|`mauricer`
+
+The BEAST2 tool suite has -at the moment- more functionality than `babette`.
+The `babette` GitHub page has a place where feature requests can be and
```
```suggestion
The [`babette` GitHub page](https://github.com/ropensci/babette) has a place where feature requests can be and
```
link to the page

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+To set a good example, I wanted my code reviewed.
+
+I enjoy having my code reviewed. Even when trying to
+write examplary and spotless code, reviewers usually 
```
```suggestion
write exemplary and spotless code, reviewers usually 
```
examplary -> exemplary

richelbilderbeek commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,202 @@
+underlying the alignment! You can see the inference has
+problems recovering it. Part of my research is to measure
+exactly this error. Therefore, I would not only need
+thousands of posteriors, but also thousands of posteriors.
```
```suggestion
thousands of phylogenies, but also thousands of posteriors.
```

stefaniebutland commented:
> what is the preferred order: getting the package CRAN first, or having a blog post first? (babette is not yet on CRAN)

That is your choice. If you're anxious to get it published, we could publish 2019-02-05. If you would like to wait until it's on CRAN, then we can keep the post as a pull request and only then set a publication date.

> I will process the feedback on Friday before 18:00 (Amsterdam time zone, UTC +1:00)

Once that's done, I will set a time to review the updates. Thank you

richelbilderbeek commented:
As promised, I processed most of your feedback [here](https://github.com/ropensci/babette/issues/56). 

I will need another week to send emails to the contributors if and how they want to be mentioned.

I will then finish the process on Friday February 8th before 18:00 (Amsterdam time zone, UTC +1:00). OK?

stefaniebutland commented:
Thank you for the update @richelbilderbeek. 
I notice that the build is now failing from an earlier commit. Let me know if you aren't able to resolve this and we can help.

richelbilderbeek commented:
@stefaniebutland: as promised, I've updated the blog post and fixed the build again.

I would like to delay the actual post until the package is on CRAN. Is that OK? I estimate that will be April 2019. Would that work out? Sure, I can strain myself to speed up if really needed :racehorse: .

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,208 @@
+### Acknowledgements
+
+Yacine Ben Chehida and Paul van Els supplied the first use cases.
+Reviews were provided by David Winter (@dwinter) 
```
```suggestion
Reviews were provided by [David Winter](https://ropensci.org/authors/david-winter/) 
```
link to David Winter author page

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,208 @@
+Yacine Ben Chehida and Paul van Els supplied the first use cases.
+Reviews were provided by David Winter (@dwinter) 
+and Joëlle Barido-Sottani (@bjoelle), 
+with a contribution from Guangchuang Yu (@GuangchuangYu).
```
```suggestion
with a contribution from [Guangchuang Yu](https://guangchuangyu.github.io/).
```

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,208 @@
+with a contribution from Guangchuang Yu (@GuangchuangYu).
+Noam Ross (@noamross) handled the review process. 
+Raphael Scherrer and Jana Riederer helped by submitting bug reports. 
+Not only was code reviewed, this (non-code) blog post was reviewed 
```
Thank you for the acknowledgement. It is not necessary to do this for me or for the editor handling your review though. In this case "it's the thought that counts".

richelbilderbeek replied:
OK, I will remove the line. Then here: thanks you for your help :+1:

richelbilderbeek replied:
... also Noah's :rainbow:

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,208 @@
+
+Yacine Ben Chehida and Paul van Els supplied the first use cases.
+Reviews were provided by David Winter (@dwinter) 
+and Joëlle Barido-Sottani (@bjoelle), 
```
```suggestion
and [Joëlle Barido-Sottani](https://github.com/bjoelle), 
```

richelbilderbeek commented:
@stefaniebutland: thanks for the excellent feedback! I am in the process of getting the packages on CRAN as we speak...

stefaniebutland commented:
That figure of posterior trees looks really neat.
Post looks good. I'll suggest a publication date once you let me know it's on CRAN
Cheers

stefaniebutland commented:
Hello @richelbilderbeek. I'm following up on your post status. Any closer to getting `babette` on CRAN?

richelbilderbeek commented:
Thanks for asking!  Yes I am getting closer, approx halfway.

To be precise, I've got 2 out of 5 packages on CRAN now. 3rd is re-submitted. 

If you want, I can post every package's acceptance, else I'll post when all are accepted.

stefaniebutland commented:
> I've got 2 out of 5 packages on CRAN now. 3rd is re-submitted.

Congratulations :-). Great to hear it's in process.
I saw this issue https://github.com/ropensci/babette/issues/8, so if you update that I can check there.

Did you receive an invitation to rOpenSci Slack? If not, I can send one to your email address I found on your website. People there can be very helpful with things like CRAN submission.

richelbilderbeek commented:
> I saw this issue ropensci/babette#8, so if you update that I can check there.

Updated!

> Did you receive an invitation to rOpenSci Slack?

Yes, up until now all CRAN feedback was crystal clear, else I consider taking a look there.

richelbilderbeek commented:
Yahoo, got the third of five packages on CRAN... I expect it to go downhill from here....

richelbilderbeek commented:
Still in the process of getting the fourth (out of 5 packages) on CRAN. Submitted `mauricer` in June, got feedback two weeks ago, resubmitted today, 11th Nov...

richelbilderbeek commented:
`mauricer` (the 4th out of 5 packages) [is on CRAN now](https://cran.r-project.org/web/packages/mauricer/), will submit the final package this week :relieved:

stefaniebutland commented:
So close now @richelbilderbeek ! congratulations

I don't currently see you in rOpenSci Slack (helpful for lots of questions including about CRAN submission). I'm sorry if we missed sending you an invitation. Is `richel@richelbilderbeek.nl` the email I should use to invite you?

richelbilderbeek commented:
>  Is richel@richelbilderbeek.nl the email I should use to invite you?

That's my email alright! AFAIK, I have already been invited.

richelbilderbeek commented:
After close to a year, `babette` is on CRAN! Because the `babette` interface has new superior alternatives, I will check the blog post this week and give the thumbs up. Will do so before Friday 23:59.

stefaniebutland commented:
Outstanding news @richelbilderbeek ! Congratulations.

I will do a final review and assign a publication date after you have given your thumbs up.

richelbilderbeek commented:
@stefaniebutland: I've checked the R code and everything is still valid. Here the thumbs up :+1:

stefaniebutland commented:
@richelbilderbeek I will review the post for content by end of day Mon Jan 20.

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+---
+slug: babette
+title: Call BEAST2 from R
+package_version: 2.0
```
update `package_version`?

https://docs.ropensci.org/babette/ on 2020-01-20 says v2.1.1

richelbilderbeek replied:
Done :+1:

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+---
+slug: babette
+title: Call BEAST2 from R
```
Most readers won't know what BEAST2 is, so they might not read. I suggest revising title to be more informative. Something like "Call BEAST2 for Bayesian evolutionary analysis from R"?

richelbilderbeek replied:
Thanks, I followed your suggestion :+1:

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+package_version: 2.0
+authors:
+  - Richèl J.C. Bilderbeek
+date: 2019-01-06
```
```suggestion
date: 2020-01-28
```

set publication date (happy anniversary!)

richelbilderbeek replied:
Updated again to today.

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+are phylogenies. For example, this a phylogeny, with four species,
+eloquently named '1', '2', '3' and '4':
+
+![A phylogeny with four species](img/blog-images/2019-01-06-babette/phylogeny.png)
```
We now recommend using [Hugo shortcodes](https://gohugo.io/content-management/shortcodes/#use-hugos-built-in-shortcodes) to add images. 

Example to be modified: `{{< figure src = "/img/blog-images/2019-06-04-post-filename/name-of-image.png" width = "200" alt = "this is alternative text that describes the image" >}}`

We now recommend that every image be accompanied by alternative text (alt="some text") to make it more accessible to people with disabilities and provide a better user experience for everyone. The alternative text should convey the meaning or content that is displayed in the image. For information on what should go in alternative text, refer to [this tutorial](https://www.w3.org/WAI/tutorials/images/informative/).

richelbilderbeek replied:
Updated all images :+1:

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+
+### References
+
+[^1]: Bilderbeek, Richel JC, and Rampal S. Etienne. "babette: BEAUti 2, BEAST 2 and Tracer for R." Methods in Ecology and Evolution (2018). [doi.org/10.1111/2041-210X.13032](https://doi.org/10.1111/2041-210X.13032)
```
In preview (https://deploy-preview-415--ropensci.netlify.com/blog/2019/01/06/babette/#fn:1), looks like `1111/2041-` in DOI is interpreted as math. Can you fix by escaping the `/`?

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+of how species form new species. Instead of working on a species'
+individuals, I work on species as evolutionary lineages.
+A good way to show the evolutionary relationships between species
+are phylogenies. For example, this a phylogeny, with four species,
```
```suggestion
are phylogenies. For example, this is a phylogeny, with four species,
```

added missing word "is"

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+a DNA alignment. Such DNA alignments are where most researchers start from.
+Here is such an alignment, of four (eloquently named) species
+and their aligned DNA sequences, in which a different color represents a
+different nucleotide:
```
```suggestion
different nucleotide (an individual element of a DNA sequence):
```

I suggest a (generic?) definition of "nucleotide" because many readers won't know, and the rest of your explanation is so beautifully clear.

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+![Posterior trees of four species](img/blog-images/2019-01-06-babette/densitree.png)
+
+Spoiler: actually, the phylogeny shown was the true phylogeny 
+underlying the alignment! You can see the inference has
```
After spoiler, it would help to briefly say how we "can see the inference has problems recovering it". That is, note how the density of lines in the image is interpreted. (Hope I am not making this more complicated than it should be. You are welcome to disagree with this or any suggested changes)

richelbilderbeek replied:
Agreed. I hope you enjoy the improvement. Feel free to make me clarify further if you think it would be better.

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+problems recovering it. Part of my research is to measure
+exactly this error. Therefore, I would not only need
+thousands of phylogenies, but also thousands of posteriors.
+There was a practical reason to make this go smooth!
```
```suggestion
There was a practical reason to make this go smoothly!
```

grammar edit

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+DensiTree is another beginner-friendly program with a graphical
+user interface to load the phylogenies of a posterior and plot these. 
+
+Again, I would need thousands of these posteriors.
```
Can you extrapolate to number of mouse clicks too?

(not required)

richelbilderbeek replied:
Added :+1:

stefaniebutland commented on content/blog/2019-01-06-babette.md:
```diff
@@ -0,0 +1,205 @@
+  - Bayesian
+---
+
+`babette` [^1] is a package to work with BEAST2 [^2], 
```
Remove backticks from all package names. Our formatting rules have changed in the last year. For formatting of package names, functions, and code, follow the [tidyverse style guidance](https://style.tidyverse.org/documentation.html#r-code).

richelbilderbeek replied:
Done :+1:

richelbilderbeek commented:
OK, most layout changes work nice on [https://deploy-preview-415--ropensci.netlify.com/blog/2020/01/22/babette/](https://deploy-preview-415--ropensci.netlify.com/blog/2020/01/22/babette/). Will fix reference weirdness (see below) now.

![weird](https://user-images.githubusercontent.com/2098230/72987236-435c8580-3dea-11ea-9933-754742c8192d.png)

richelbilderbeek commented:
Hi @stefaniebutland, I processed all your feedback and am ready for more, if any :+1:

stefaniebutland commented:
Thank you @richelbilderbeek.

I made a few final fixes, but they can be found in this [new pull request](https://github.com/ropensci/roweb2/pull/599). (Sorry, I meant to add them to yours but I was trying something new to me.)

changes:
- YAML `date: 2020-01-28` gets set to publication date, not current date
- renamed blog-images folder to 2020-01-28-babette and set paths in post to this folder
- adjusted width in `width = "200"` to suit each image 
- added tag `babette`

I'll ask my colleagues for advice on fixing the reference formatting so it can be an active link. I'll fix it when I get a solution.

No more edits needed on your end. I'll note here when the ref formatting is fixed. Thank you for your work on this. I will publish and tweet on Jan 28.

richelbilderbeek commented:
@stefaniebutland: thanks you some much, also for having the same perseverance :+1:

I look forward to our next rOpenSci blog post :rainbow:

maelle commented:
I'm very glad to be able to read this post since I'm handling the `mcbette` submission :-)

maelle commented:
A few non-asked for remarks whilst I'm installing all mcbette dependencies :wink: 

In the blog post, I couldn't find a link to

* the software review thread https://github.com/ropensci/software-review/issues/209

* the official website https://docs.ropensci.org/babette/ (ICYMI @richelbilderbeek , see https://ropensci.org/technotes/2019/06/07/ropensci-docs/)

Would it make sense to mention you've submitted a further package for peer-review, and also whether you're intending to add even more R packages to the stack?

stefaniebutland commented:
Addressed 2 of @maelle's comments in the other pull request https://github.com/ropensci/roweb2/pull/599#issuecomment-578571623

maelle commented:
I was a lazy reader hoping to be able to click from a single place 😂

richelbilderbeek commented:
@maelle: sorry for the slower response. Everything is OK now? If no, let me know :+1:

maelle commented:
The point that's not in @stefaniebutland's corrected version of your post is "Would it make sense to mention you've submitted a further package for peer-review, and also whether you're intending to add even more R packages to the stack?" (but it's just a suggestion!).

maelle commented:
(besides, I don't think the answer was slow, and I was aware I was posting on a week-end!)

richelbilderbeek commented:
Uh, AFAICS the post is online. Would it still make sense to refer to my next package?

 * If yes: will fix tomorrow/Wednesdat
 * If no: sorry for being slow

stefaniebutland commented:
Post is published: https://ropensci.org/blog/2020/01/28/babette/ and tweeted https://twitter.com/rOpenSci/status/1222190847900274688 

I think it's ok to leave as is, but am happy to make the edit if you want to provide text. I'll close this PR once you confirm your decision @richelbilderbeek 

No need to apologize for timing. You have been very prompt in submission and revisions.

richelbilderbeek commented:
@stefaniebutland: Thanks, you did the right thing :+1:

It was great working with you, thanks so much!
